apollo-router/src/router_factory.rs

```diff
@@ -631,7 +632,22 @@ pub(crate) async fn create_plugins(
                         // Only the telemetry plugin should have access to the full configuration
                         full_config = configuration.validated_yaml.clone();
                     }
-                    add_plugin!(name.to_string(), factory, plugin_config, full_config);
+
+                    // If the license has an allowed_features claim, we know we're using a pricing
+                    // plan with a subset of allowed features
+                    if let Some(allowed_features) = $license.get_allowed_features() {
+                        if allowed_features.contains(&AllowedFeature::from($name)) {
+                            add_plugin!(name.to_string(), factory, plugin_config, full_config);
+                        } else {
+                            tracing::warn!(
+                                "{name} is a restricted feature that requires a license"
```

Comment:
So we are not initialising these plugins, right? the message should probably have something alluding to that, for example:
```suggestion
                                "{name} plugin is not registered, {name} is a restricted feature that requires a license"
```